File: Code/java/palindrome.java
import java.util.Scanner;

class Palindrome
{
    public static void main(String[] args)
    {
        String input, result = "Palindrome";
        int length, i;
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter Input: ");
        input = scanner.nextLine();

        length = input.length();
        for (i = 0; i < length / 2; i++)
        {
            if (input.charAt(i) != input.charAt(length - i - 1))
            {
                result = "Not Palindrome";
                break;
            }
        }
        System.out.println(result);

        scanner.close();
    }
}

File: Code/cpp/palindrome.cpp
#include<iostream>
#include<string.h>
using namespace std;
int main()
{
    char input[25];
    string result = "Palindrome";
    int length, i;
    
    cout << "Enter Input: ";
    cin >> input;

    length = strlen(input);
    for (i = 0; i < length / 2; i++)
    {
        if (input[i] != input[length - i - 1])
        {
            result.replace(0, 14, "Not Palindrome");
            break;
        }
    }
    cout << result;
    return 0;
}